packages/app/src/runs/CloudConnectButton.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <Button
+    :class="props.class"
+    :prefix-icon="isLoggedIn ? ChainIcon : UserIcon"
+    prefix-icon-class="icon-dark-white icon-light-transparent"
+    @click="openConnection"
+  >
+    {{ isLoggedIn ? t('runs.connect.buttonProject') : t('runs.connect.buttonUser') }}
+  </Button>
+  <LoginModal
+    v-model="isLoginOpen"
+    :gql="props.gql"
+  />
+  <CloudConnectModals
+    v-if="isProjectConnectOpen"
+    :show="isProjectConnectOpen"
+    :gql="props.gql"
+    @cancel="isProjectConnectOpen = false"
+    @success="isProjectConnectOpen = false; emit('success')"
+  />
+</template>
+
+<script lang="ts" setup>
+import { computed, ref } from 'vue'
+import { gql } from '@urql/vue'
+import UserIcon from '~icons/cy/user-outline_x16.svg'
+import ChainIcon from '~icons/cy/chain-link_x16.svg'
+import Button from '@cy/components/Button.vue'
+import CloudConnectModals from './modals/CloudConnectModals.vue'
+import LoginModal from '@cy/gql-components/topnav/LoginModal.vue'
+import { useI18n } from '@cy/i18n'
+import type { CloudConnectButtonFragment } from '../generated/graphql'
+
+const { t } = useI18n()
+
+gql`
+fragment CloudConnectButton on Query {
+  ...CloudConnectModals
+  ...LoginModal
+}
+`
+
+const emit = defineEmits<{
```

Comment:
Is the `success` event used anywhere?

Reply:
Ooh, good question. It looks like this emits out to RunsConnect.vue (where this code was previously) when then needs to emit out to its parent. I'll update this. (Still getting used to Vue/Vue 3)

Reply:
I guess I'm not sure if the settings page needs to have similar functionality as well?

Reply:
Ahh it was a good catch then lol. I'll take another look and verify